src/lib/albertcore/src/albert/mainwindow/mainwindow.cpp

```diff
@@ -227,6 +227,12 @@ void MainWindow::setVisible(bool visible) {
 /** ***************************************************************************/
 void MainWindow::toggleVisibility() {
    setVisible(!isVisible());
+   // albert might fail to grab keyboard focus if you press keyboard immediately right after HOTKEY was triggered.
+   QTimer::singleShot(1, this, &MainWindow::regrabFocus);
```

Comment:
@ManuelSchneid3r 
Yes, I did, in fact, I have been using this code for quiet a while. 
To my problem, yes, Albert was visible, on top of all windows, only it didn't have the keyboard for ever, unless I toggle Alert twice without immediate typing. I was using Linux Mint 18.1 (ubuntu base).
It's not a perfect fix, as some characters still go to previous window before the fix kicks in, but at least I don't need to toggle Albert twice...